File: src/Connect.Api/ClientTokens/Values/ClientDto.cs
namespace Connect.ClientTokens;

public sealed record ClientDto(
    string Token,
    string Name,
    string? Email,
    string? PhoneNumber,
    List<PropertyRecord>? SellRecords,
    List<PropertyRecord>? BuyRecords,
    List<SoldInventoryDto>? SoldInventories,
    List<NewInventoryDto>? NewInventories,
    ClientPayment? Payment)
{
    public static ClientDto ToDto(Client client, List<SoldInventoryDto> soldInventories, List<NewInventoryDto> newInventories)
    {
        return new ClientDto(
            client.Token.GetToken(),
            client.Name.Value,
            client.EmailId?.Value,
            client.PhoneNo,
            client.SellRecords,
            client.BuyRecords,
            soldInventories,
            newInventories,
            client.Payment);
    }
}

File: frontend/src/core/models/ClientDto.ts
import { SoldInventoryDto, NewInventoryDto } from "./InventoryDto";
export interface PropertyRecord {
    id: string;
    originalrate: number;
    draftRate?: number; 
    approvedRate?: number; 
    paymentPlan?: string;
  }
  
  export interface ClientPayment {
    amountPaid: number;
    paymentMode?: string;
    paymentId: number;
    channelPartnerId?: string;
    customChannelPartnerName?: string;
    customChannelPartnerNumber?: string;
  }

  export interface ClientDto {
    id: string;
    name: string;
    email?: string;
    phoneNumber?: string;
    sellRecords?: PropertyRecord[];
    buyRecords?: PropertyRecord[];
    soldInventories?: SoldInventoryDto[];
    newInventories?: NewInventoryDto[];
    payment?: ClientPayment;
  }
  